### PR TITLE
✨ Consent Metadata macros in ads RTC and amp-analytics

### PR DESCRIPTION
**Pull request by micajuine-ho**
For #30006

### CODE REVIEW

File: extensions/amp-a4a/0.1/test/test-amp-a4a.js [hunk 1]
```diff
@@ -2549,7 +2551,13 @@ describes.realWin('no signing', {amp: true}, (env) => {
             a4a = new MockA4AImpl(a4aElement);
             consentString = 'test-consent-string';
             gdprApplies = true;
-            consentMetadata = {gdprApplies};
+            additionalConsent = 'abc123';
+            consentStringType = 1;
+            consentMetadata = {
```
Comment (zhouyx) [inline extensions/amp-a4a/0.1/test/test-amp-a4a.js:2556]: nit: can save a few lines of code by inlining `gdprApplies`, `addtionalConsent` and `consentStringType` value.

File: extensions/amp-analytics/0.1/variables.js [hunk 2]
```diff
@@ -273,6 +273,8 @@ export class VariableService {
       'COOKIE': (name) =>
         cookieReader(this.ampdoc_.win, dev().assertElement(element), name),
       'CONSENT_STATE': getConsentStateStr(element),
+      'CONSENT_METADATA': (key) =>
+        getConsentMetadataValue(element, dev().assertString(key)),
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/variables.js:277]: I believe not including the key is a user error?

File: extensions/amp-analytics/0.1/test/test-requests.js [hunk 3]
```diff
@@ -610,6 +610,21 @@ describes.realWin('Requests', {amp: 1}, (env) => {
...
+      'baseUrl':
+        'r1&$CONSENT_METADATA(gdprApplies)test&${consentMetadata(additionalConsent)}test2',
+    };
+    const handler = createRequestHandler(r, spy);
+    const expansionOptions = new ExpansionOptions({
+      'consentMetadata': 'CONSENT_METADATA',
+    });
+    handler.send({}, {}, expansionOptions);
+    await macroTask();
+    expect(spy).to.be.calledWith('r1&test&test2');
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-requests.js:625]: The test doesn't seem to include the expanded value? Is this expected

Comment (micajuine-ho) [reply extensions/amp-analytics/0.1/test/test-requests.js:625]: Yes, that's expected. Will also add a test for expanding it.

File: extensions/amp-analytics/0.1/test/test-variables.js [hunk 4]
```diff
@@ -444,6 +444,25 @@ describes.fakeWin('amp-analytics.VariableService', {amp: true}, (env) => {
...
+              'gdprApplies': true,
+              'additionalConsent': 'abc123',
+              'consentStringType': 1,
+            };
+          },
+        })
+      );
+
+      return check(
+        'CONSENT_METADATA(gdprApplies)&CONSENT_METADATA(additionalConsent)&CONSENT_METADATA(consentStringType)',
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-variables.js:461]: worth adding a `CONSENT_METADATA(invalid_key)`, and check that it resolves to null.

**Review by zhouyx — APPROVED** [review]
LGTM!

**Review by rsimha — APPROVED** [review]

**Review by lannka — APPROVED** [review]

**Pull request merged**